lasy/utils/laser_utils.py

```diff
@@ -873,3 +873,167 @@ def import_from_z(dim, grid, omega0, field_z, z_axis, z0=0.0, t0=0.0, backend="N
+    r"""
+    Calculate the laser waist.
+
+    Parameters
+    ----------
+    dim : string
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
+
+    Return
+    ----------
+    sigma: Standard deviation of a**2 in m
+    """
+    field = grid.get_temporal_field()
+    if dim == "xyt":
+        Nx, Ny, Nt = field.shape
+        A2 = (np.abs(field[Nx // 2 - 1, :, :]) ** 2).sum(-1)
+        ax = grid.axes[1]
+    else:
+        A2 = (np.abs(field[0, :, :]) ** 2).sum(-1)
+        ax = grid.axes[0]
+        if ax[0] > 0:
+            A2 = np.r_[A2[::-1], A2]
+            ax = np.r_[-ax[::-1], ax]
+        else:
+            A2 = np.r_[A2[::-1][:-1], A2]
+            ax = np.r_[-ax[::-1][:-1], ax]
+
+    sigma = 2 * np.sqrt(np.average(ax**2, weights=A2))
+
+    return sigma
+
+
+def get_STC(dim, grid, k0):
+    r"""
+    Calculate the spatio-temporal coupling factors of the laser.
+
+    Parameters
+    ----------
+    dim : string
+        Dimensionality of the array. Options are:
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    grid : a Grid object.
+        It contains an ndarray (V/m) with
+        the value of the envelope field and the associated metadata
+        that defines the points at which the laser is defined.
+
+    k0 : scalar
+        Wavenumber of the field
+
+    Return
+    ----------
+    STC_fac : dict of floats
+        A dictionary of floats corresponding to the STC factors. The keys are:
+            Phi2: Group-delayed dispersion in :math:`\Phi^{(2)}=d(\omega_0)/dt`
+            phi2: Group-delayed dispersion in :math:`\phi^{(2)}=dt_0/d(\omega)`
+            nu_x, nu_y: Spatio-chirp in :math:`\nu=d(\omega_0)/dx`
+            zeta_x, zeta_y: Spatio-chirp in :math:`\zeta=dx_0/d(\omega_0)`
+            beta_x, beta_y: Angular dispersion in :math:` \beta = d\theta_0/d\omega`(Important note:
+                  for now beta is only correct when zeta and phi2 are 0!)
+            pft_x, pft_y: Pulse front tilt in :math:` p=dt/dx`
+    All those above units and definitions are taken from
+    `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.
+    """
+    tau = 2 * get_duration(grid, dim)
+    w0 = get_w0(grid, dim)
+
+    # Initialise the returned dictionary
```

Comment:
US English was used above, and is the norm in LASY.
```suggestion
    # Initialize the returned dictionary
```